### PR TITLE
Update governance documentation

### PR DESCRIPTION
**Pull request by ehuss**
This updates the governance documentation to reflect the changes as a result of [RFC 3392](https://rust-lang.github.io/rfcs/3392-leadership-council.html). This also replaces mentions of the core team with the leadership council. There are several commits which should explain each change. The intent here is for the official governance documentation lives here as a permanent home.

Closes https://github.com/rust-lang/rust-forge/issues/694
Closes https://github.com/rust-lang/leadership-council/issues/14

### CODE REVIEW

**Review by rylev — APPROVED** [review]
Thanks for putting this together!!!

File: src/governance/council.md
Comment (ehuss) [reply src/governance/council.md]: Thanks. I'm not entirely sure why the RFC authors framed it that way, but I agree it seems like it doesn't fit.

File: src/governance/council.md
Comment (ehuss) [reply src/governance/council.md]: "Council" is a short form of the Leadership Council which is a short form of the Rust Leadership Council. I added a commit to consistently use Council.

**Comment by ehuss** [comment]
@rylev Just checking if there is anything else you'd like to see before merging this or if it is waiting on anything?

**Comment by rylev** [comment]
@ehuss nope  - I was expecting you to merge it, but I can if you want me to.

**Comment by ehuss** [comment]
I think you are the only one with permission to merge things here (and Mark, but I'm not really clear who else has access).

**Review by rylev — APPROVED** [review]

**Comment by rylev** [comment]
We should discuss giving merge permissions at least to the librarians committee of the Council.

**Pull request merged**
